### PR TITLE
fix OPENROWSET with numeric codepage as string

### LINKED ISSUE #63: Codepage='65001' or any numeric value is throwing Incorrect Syntax error in OPENROWSET query
While running the below script in sqlproj using Visual Studio 17.9 preview 1, I am getting the error 'Incorrect syntax near '65001' while the same query runs successfully in SSMS 19.

SELECT *
FROM OPENROWSET(
    BULK 'path_to_csv',
	SINGLE_BLOB,
    CODEPAGE = '65001' -- UTF-8 encoding
) AS t;

### PR DESCRIPTION
**Pull request by lBilali**
This pull request fixes #63 

since when codepage is a number no check is done if the value is a valid codepage this change does the same when the code page is a number as string

### CODE REVIEW

**Comment by lBilali** [comment]
@microsoft-github-policy-service agree

**Review by llali — CHANGES_REQUESTED** [review]

File: SqlScriptDom/Parser/TSql/TSql100.g [hunk 1]
```diff
@@ -3524,7 +3524,9 @@ simpleBulkInsertOptionWithValue returns [LiteralBulkInsertOption vResult = Fragm
             {
                 vResult.OptionKind = BulkInsertStringOptionsHelper.Instance.ParseOption(tOption, SqlVersionFlags.TSql100);
                 UpdateTokenInfo(vResult, tOption);
-                if (vResult.OptionKind == BulkInsertOptionKind.CodePage) // Check for code page string constants
+                // Check for code page string constants. 
+                // If codepage is int just use the value as is. Same as above when value is integerOrNumeric
+                if (vResult.OptionKind == BulkInsertOptionKind.CodePage && !int.TryParse(vValue.Value, out int value))
```
[the remark this answers is not in the document]
Comment (lBilali) [reply SqlScriptDom/Parser/TSql/TSql100.g:3529]: I added a test case for CODEPAGE = '866' on the first commit but now I did one extra test case for CODEPAGE = '65001' as well
and both work

BTW I verifyed that it fails if there is a not number value other then 'ACP', 'OEM' and 'RAW' and it failes as expected.
NO test case added on that though

**Review by zijchen — APPROVED** [review]

**Review by llali — APPROVED** [review]

**Comment by jashwood** [comment]
I'm still getting this in the latest version of VS. The OPENROWSET documentation clearly has single quotes around the value expected, making it look like it should expect a string

**Comment by dzsquared** [comment]
You're right @jashwood, I'm also seeing an old version of scriptdom in VS. Checking with the team if the update can be picked into the next preview release.

**Pull request merged**